UnityProject/Assets/Scripts/UnityMarines/Items/Util/Supply/ASRSBeacon.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Mirror;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Objects.TGMC
+{
+	public class ASRSBeacon : MonoBehaviour, ICheckedInteractable<HandActivate>
+	{
+		private bool active = false;
+
+		private SpriteHandler spriteHandler;
+
+		private RegisterTile registerTile;
+
+		private static HashSet<ASRSBeacon> activeBeacons = new HashSet<ASRSBeacon>();
+
+		[field: SerializeField]
+		public string Name { get; private set; }
+
+		#region LifeCycle
+
+		private void Awake()
+		{
+			spriteHandler = GetComponentInChildren<SpriteHandler>();
+			registerTile = GetComponent<RegisterTile>();
+		}
+
+		private void OnDisable()
+		{
+			DeactivateBeacon(true);
+		}
+
+		[RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.SubsystemRegistration)]
+		private static void StaticClear()
+		{
+			activeBeacons.Clear();
+		}
+
+		#endregion
+
+		[Server]
+		public void ActivateBeacon()
+		{
+			if(active) return;
+
+			active = true;
+			//spriteHandler.ChangeSprite(0); no alternate sprites rn
+
+			activeBeacons.Add(this); //Shouldn't need to add test for this as should auto fail if already exists cause hashset
+		}
+
+		[Server]
+		public void DeactivateBeacon(bool destroyed = false)
+		{
+			if(active == false) return;
+			active = false;
+
+			//spriteHandler.ChangeSprite(1); dont have multiple textures for this beacon rn
+
+			if (activeBeacons.Contains(this)) activeBeacons.Remove(this);
+		}
+
+		[Server]
+		public Vector3 CurrentBeaconPosition()
+		{
+			return registerTile.WorldPositionServer;
+		}
+
+		public static HashSet<ASRSBeacon> GetActiveBeacons()
+		{
+			return activeBeacons;
+		}
+
+		public bool WillInteract(HandActivate interaction, NetworkSide side)
+		{
+			if (DefaultWillInteract.Default(interaction, side) == false) return false;
```

Comment:
You can just return `DefaultWillInteract.Default(interaction, side)`directly since there is nothing else that needs to be checked